src/server/zset_family.cc

```diff
@@ -265,6 +265,10 @@ void IntervalVisitor::operator()(ZSetFamily::TopNScored sc) {
 }
 
 void IntervalVisitor::ActionRange(unsigned start, unsigned end) {
+  if (params_.limit == 0)
+    return;
+  start += params_.offset;
+  end = static_cast<uint32_t>(min(1ULL * start + params_.limit - 1, 1ULL * end));
```

Comment:
what line 271 does?

Reply:
Calculates new end if limit is set (< INT MAX), added a comment above